### PR TITLE
[continuous-integration] initial migration to GitHub actions

### PR DESCRIPTION
**Pull request by jwhui**
GitHub Actions offers many more parallel build jobs than Travis CI.

Also cleaned up scripts in the transition.

This should reduce the check time to around 30 minutes.

### CODE REVIEW

File: script/check-arm-build-autotools [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    ./bootstrap
+    make -f examples/Makefile-nrf52833 "${options[@]}" DISABLE_TRANSPORTS=1
+}
+
+build_nrf52840()
+{
+    local options=(
+        "BORDER_AGENT=1"
+        "BORDER_ROUTER=1"
+        "COAP=1" "COAPS=1"
```
Comment (bukepo) [inline script/check-arm-build-autotools]:
```suggestion
        "COAP=1"
        "COAPS=1"
```

File: script/check-simulation-build-autotools [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        "-DOPENTHREAD_CONFIG_SOFTWARE_ENERGY_SCAN_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_SOFTWARE_RETRANSMIT_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_TMF_NETWORK_DIAG_MTD_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_UDP_FORWARD_ENABLE=1"
+    )
+
+    reset_source
+    export CPPFLAGS="${options[*]} -DOPENTHREAD_CONFIG_LOG_LEVEL=OT_LOG_LEVEL_NONE"
+    ./bootstrap
```
Comment (bukepo) [inline script/check-simulation-build-autotools]: bootstrap costs a lot of time. How about just removing `build` and `output` and `tmp` in reset_source?

**Comment by bukepo** [comment]
@jwhui Where to check the GitHub actions results?

**Comment by jwhui** [comment]
> @jwhui Where to check the GitHub actions results?

I don't think you will see them until the PR is merged.

**Comment by jwhui** [comment]
For the time being, we will redundantly run both Travis CI and GitHub Actions. Once we are comfortable with GitHub Actions, we can start to disable Travis CI tests.

**Comment by jwhui** [comment]
> > @jwhui Where to check the GitHub actions results?
> 
> I don't think you will see them until the PR is merged.

You can see the GitHub Action status here:
- https://github.com/openthread/openthread/action

And on a PR here:
- #4730

**Pull request merged**